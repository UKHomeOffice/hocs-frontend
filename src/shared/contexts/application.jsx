import React, { Component } from 'react';
import PropTypes from 'prop-types';
import types from './actions/types.jsx';

const Context = React.createContext();

const reducer = (state, action) => {
    // TODO: REMOVE
    /* eslint-disable-next-line  no-console*/
    console.log(`ACTION: ${action.type} PAYLOAD: ${JSON.stringify(action.payload)}`);
    // ------------
    switch (action.type) {
    case types.UPDATE_FORM:
        return {
            ...state, form: action.payload
        };
    case types.UPDATE_FORM_DATA:

        return {
            ...state, form: {
                ...state.form, data: state.form && state.form.data ? { ...state.form.data, ...action.payload } : { ...action.payload }
            }
        };
    case types.UPDATE_FORM_ERRORS:
        return {
            ...state, form: {
                ...state.form, errors: action.payload
            }
        };
    case types.UPDATE_LOCATION:
        return { ...state, location: action.payload };
    case types.CANCEL:
        return { ...state, redirect: '/', form: null };
    case types.REDIRECT:
        return { ...state, redirect: action.payload, form: null };
    case types.REDIRECTED:
        return { ...state, redirect: null };
    case types.SET_ERROR:
        return { ...state, error: action.payload };
    case types.UNSET_ERROR:
        return { ...state, error: null };
    case types.UNSET_FORM:
        return { ...state, form: null };
    case types.UPDATE_API_STATUS:
        return { ...state, apiStatus: action.payload };
    case types.CLEAR_API_STATUS:
        return { ...state, apiStatus: null };
    case types.UPDATE_PAGE_META:
        return { ...state, page: { ...action.payload } };
    case types.UPDATE_DASHBOARD:
        return { ...state, dashboard: action.payload };
    case types.CLEAR_DASHBOARD:
        return { ...state, dashboard: null };
    case types.UPDATE_WORKSTACK:
        return { ...state, workstack: action.payload };
    case types.CLEAR_WORKSTACK:
        return { ...state, workstack: null };
    default:
        // TODO: Remove
        /* eslint-disable-next-line  no-console*/
        console.warn('Unsupported action');
    }
};

export class ApplicationProvider extends Component {
    constructor(props) {
        super(props);
        this.state = {
            ...props.config,
            apiStatus: null,
            dispatch: action => {
                try {
                    this.setState(state => reducer(state, action));
                    return Promise.resolve();
                } catch (error) {
                    return Promise.reject(error);
                }
            }
        };
    }

    render() {
        const { state, props: { children } } = this;
        return (
            <Context.Provider value={state}>
                {children}
            </Context.Provider>
        );
    }

}

ApplicationProvider.propTypes = {
    children: PropTypes.node,
    config: PropTypes.object
};

export const ApplicationConsumer = Context.Consumer;