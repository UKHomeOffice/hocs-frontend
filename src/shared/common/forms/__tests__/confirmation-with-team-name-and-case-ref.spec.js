import React from 'react';
import ConfirmationWithTeamNameAndCaseRef from '../confirmation-with-team-name-and-case-ref.jsx';
import { ApplicationProvider } from '../../../contexts/application.jsx';

describe('Panel component', () => {
    const config = {
        csrf: '__token__'
    };

    const choices = [
        {
            key: '4ef92480-922f-11eb-a8b3-0242ac130003',
            value: 'FOI BICSPI Acceptance Team'
        },
        {
            key: 'abc92480-922f-11eb-a8b3-0242ac130003',
            value: 'Another Team'
        }
    ];
    const hasHeader = true;
    const defaultData = {};
    const name = 'variableName';
    const data = {
        variableName: '4ef92480-922f-11eb-a8b3-0242ac130003'
    };

    it('should render with default props', () => {
        expect(
            render(<ApplicationProvider config={config}>
                <ConfirmationWithTeamNameAndCaseRef data={defaultData}/>
            </ApplicationProvider>)
        ).toMatchSnapshot();
    });


    it('should render with title and team name when passed', () => {
        expect(
            render(<ApplicationProvider config={config} >
                <ConfirmationWithTeamNameAndCaseRef data={data} choices={choices} name={name} label="completed" caseRef={'test-case-ref'} />
            </ApplicationProvider>)
        ).toMatchSnapshot();
    });

    it('should render with title, team name and header box when passed', () => {
        expect(
            render(<ApplicationProvider config={config} >
                <ConfirmationWithTeamNameAndCaseRef data={data} choices={choices} name={name} label="completed" caseRef={'test-case-ref'} hasHeader={hasHeader} />
            </ApplicationProvider>)
        ).toMatchSnapshot();
    });

});
