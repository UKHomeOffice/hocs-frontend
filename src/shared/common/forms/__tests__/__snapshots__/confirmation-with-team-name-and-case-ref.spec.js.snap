// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel component should render with default props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2
          class="govuk-heading-m"
        >
          Case 
           
           
        </h2>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2
        class="govuk-heading-m"
      >
        Case 
         
         
      </h2>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Panel component should render with title and team name when passed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2
          class="govuk-heading-m"
        >
          Case 
          test-case-ref
           
          completed
           
          FOI BICSPI Acceptance Team
        </h2>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2
        class="govuk-heading-m"
      >
        Case 
        test-case-ref
         
        completed
         
        FOI BICSPI Acceptance Team
      </h2>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Panel component should render with title, team name and header box when passed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="govuk-panel govuk-panel--confirmation"
        >
          <div
            class="govuk-panel__body"
          >
            Case 
            completed
            : 
            FOI BICSPI Acceptance Team
          </div>
        </div>
        <h2
          class="govuk-heading-m"
        >
          Case 
          test-case-ref
           
          completed
           
          FOI BICSPI Acceptance Team
        </h2>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="govuk-panel govuk-panel--confirmation"
      >
        <div
          class="govuk-panel__body"
        >
          Case 
          completed
          : 
          FOI BICSPI Acceptance Team
        </div>
      </div>
      <h2
        class="govuk-heading-m"
      >
        Case 
        test-case-ref
         
        completed
         
        FOI BICSPI Acceptance Team
      </h2>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
